source/core/TDCodecVia.cpp

```diff
@@ -959,7 +959,7 @@ Int32 TDViaParser::ParseArrayData(TypedArrayCoreRef pArray, void* pFirstEltInSli
                 _string.EatChar(Fmt()._itemSeparator);
                 void* pElement = elementCount < length ? pEltData : nullptr;
                 Int32 subErr;
-                if (pElement == nullptr) {
+                if (elementCount >= length) {
```

Comment:
What was happening is that pEltData was also nullptr for empty arrays, so even if elementCount was < length, we were treating it as if there were extra initializers. I change the check here which only occurs if there really are extra initializers and not just an empty array.